apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/logging/LoggingFilter.java

```diff
@@ -0,0 +1,147 @@
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Set;
+import org.jboss.resteasy.reactive.server.ServerResponseFilter;
+import org.slf4j.Logger;
+
+/** The filter for logging requests when required for debugging purposes. */
+@ApplicationScoped
+public class LoggingFilter {
+  private static final Logger logger = org.slf4j.LoggerFactory.getLogger(LoggingFilter.class);
+
+  /** The {@link ObjectMapper} to get request body from request. */
+  private final ObjectMapper objectMapper;
+  /** The request info bean. */
+  private final StargateRequestInfo requestInfo;
+  /** The configuration for logging. */
+  private final LoggingConfig loggingConfig;
+
+  /** Default constructor. */
+  @Inject
+  public LoggingFilter(
+      ObjectMapper objectMapper, StargateRequestInfo requestInfo, LoggingConfig loggingConfig) {
+    this.objectMapper = objectMapper;
+    this.requestInfo = requestInfo;
+    this.loggingConfig = loggingConfig;
+  }
+
+  /**
+   * Filter that this bean produces.
+   *
+   * @param requestContext {@link ContainerRequestContext}
+   * @param responseContext {@link ContainerResponseContext}
+   * @see https://quarkus.io/guides/resteasy-reactive#request-or-response-filters
+   */
+  @ServerResponseFilter
+  public void logRequestIfRequired(
+      ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+    if (isLoggingAllowed(requestContext, responseContext)) {
+      logger.debug(
+          buildRequestInfo(
+              requestContext, responseContext, loggingConfig.requestBodyLoggingEnabled()));
+    }
+  }
+
+  private String buildRequestInfo(
+      ContainerRequestContext requestContext,
+      ContainerResponseContext responseContext,
+      boolean logRequestBody) {
+    String requestBody = logRequestBody ? getRequestBody(requestContext) : "";
+    return String.format(
+        "REQUEST INFO :: %s %s %s %s %s",
+        requestInfo.getTenantId().orElse("tenant-unknown"),
+        responseContext.getStatus(),
+        requestContext.getMethod(),
+        requestContext.getUriInfo().getPath(),
+        requestBody);
+  }
+
+  private String getRequestBody(ContainerRequestContext requestContext) {
+    try {
+      InputStream inputStream = requestContext.getEntityStream();
+      if (inputStream == null || !inputStream.markSupported()) {
+        return "";
+      }
+      inputStream.reset();
+      return objectMapper.readTree(inputStream).toString();
+    } catch (IOException e) {
+      logger.error("Exception occurred while trying to log request info", e);
+      return "";
+    }
+  }
+
+  private boolean isLoggingAllowed(
+      ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+    if (!loggingConfig.enabled()) {
+      return false;
+    }
+    Set<String> allowedTenants =
```

Comment:
We could (and I think should) extract "contains all" setting when constructing filter, to by-pass construction of single "all" String set for the common case.

Reply:
Sorry, could you please explain it a little bit more. I'm not creating the filter up front.

Reply:
You could determine whether "all" is included as an option (or if value is missing) right when `LoggingFilter` is constructed -- it will never change. So it does not need to be checked dynamically here, can check once and store as `boolean includeAllTenants` (or whatever).

Reply:
I guess it does not matter that much, slight performance overhead. Approving.